src/js/services/highlightJSBadge.js

```diff
@@ -0,0 +1,71 @@
+window.onload = () => {
+  let codeSnippetButtons;
+
+  const options = {
+    // the selector for the badge template
+    templateSelector: '#CodeBadgeTemplate',
+
+    // base content CSS selector that is searched for snippets
+    contentSelector: 'body',
+
+    // Delay in ms used for `setTimeout` before badging is applied
+    // Use if you need to time highlighting and badge application
+    // since the badges need to be applied afterwards.
+    // 0 - direct execution (ie. you handle timing
+    loadDelay: 0,
+
+    // CSS class(es) used to render the copy icon.
+    copyIconClass: 'fa fa-copy',
+    // optional content for icons class (<i class="fa fa-copy"></i> or <i class="material-icons">file_copy</i>)
+    copyIconContent: '',
+
+    // CSS class(es) used to render the done icon.
+    checkIconClass: 'fa fa-check text-success',
+    checkIconContent: '',
+
+    // function called before code is placed on clipboard that allows you inspect and modify
+    // the text that goes onto the clipboard. Passes text and code root element (hljs).
+    // Example:  function(text, codeElement) { return text + " $$$"; }
+    onBeforeCodeCopied: function (text, codeElement) {
+      if (!codeSnippetButtons) {
+        codeSnippetButtons = document.getElementsByClassName('code-badge');
+      }
+
+      const codeSnippetPosition =
+        Array.prototype.indexOf.call(
+          codeSnippetButtons,
+          codeElement.previousSibling,
+        ) + 1;
+
+      window.AnalyticsClient.trackAction('docs-copy-code-clicked', {
+        page: location.pathname,
+        codeSnippetPosition,
+      });
+
+      return text;
+    },
+  };
+  window.highlightJsBadge(options);
+
+  if (!codeSnippetButtons) {
+    codeSnippetButtons = document.getElementsByClassName('code-badge');
+  }
+
+  const eventType = `${window.PointerEvent ? 'pointer' : 'mouse'}down`;
+  for (let i = 0; i < codeSnippetButtons.length; i++) {
```

Comment:
I think we should use a forEach here. More readable, less prone to errors.

Reply:
I can't say I've heard of the errors thing. do you have something I can read ab0ut that?

Reply:
plain for loops are faster than forEach hence why. i thought it made sense here given there could be a lot of codeSnippets on a page.

i can change it for readability reasons though.